File: Programmers/LEVEL_1/c++/체육복.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    int stu[30];
    for (int i = 0; i < n; i++)
        stu[i] = 1;

    for (int i = 0; i < lost.size(); i++)
    {
        stu[lost[i] - 1]--;
    }

    for (int i = 0; i < reserve.size(); i++)
    {
        stu[reserve[i] - 1]++;
    }

    for (int i = 0; i < n; i++)
    {
        if (stu[i] == 0)
        {
            if (stu[i + 1] == 2)
            {
                stu[i + 1]--;
                stu[i]++;
            }
            else if (stu[i - 1] == 2)
            {
                stu[i - 1]--;
                stu[i]++;
            }
        }
    }

    for (int i = 0; i < n; i++)
    {
        if (stu[i] != 0) answer++;
    }

    return answer;
}

File: Programmers/LEVEL_1/체육복.js
function solution(n, lost, reserve) {
    var answer = 0;
    let arr=[];
    for(let i=0; i<n; i++){
        arr.push(1);
    }
    
    for(let i=0; i<lost.length; i++){
        arr[lost[i]-1]--;
    }
    
    for(let i=0; i<reserve.length; i++){
        arr[reserve[i]-1]++;
    }
    
    for(let i=0; i<n; i++){
        if(arr[i]==0){
            if(arr[i+1]==2){
                arr[i]++;
                arr[i+1]--;
            }else if(arr[i-1]==2){
                arr[i]++;
                arr[i-1]--;
            }
        }
    }
    
    for(let i=0; i<n; i++){
        if(arr[i]!=0) answer++;
    }
    
    return answer;
}